### PR TITLE
Cpp non constant format as path query

### PR DESCRIPTION
**Pull request by bdrodes**
Modifying the cpp non const format query to be a path query.

### CODE REVIEW

File: cpp/ql/src/Likely Bugs/Format/NonConstantFormat.ql [hunk 1]
```diff
@@ -69,7 +69,10 @@ predicate isNonConst(DataFlow::Node node) {
   // Parameters of uncalled functions that aren't const
   exists(UncalledFunction f, Parameter p |
     f.getAParameter() = p and
-    p = node.asParameter()
+    p = node.asParameter() and 
+    // Ignore main's argv parameter as it is already considered a `FlowSource`
+    // not ignoring it will result in path redundancies
+    (f.getName() = "main" implies p.getName() != "argv")
```
Comment (MathiasVP) [inline cpp/ql/src/Likely Bugs/Format/NonConstantFormat.ql:75]: This assumes the parameter is always called `argv`. That does not have to be true.

Comment (bdrodes) [reply cpp/ql/src/Likely Bugs/Format/NonConstantFormat.ql:75]: Good point. I should've copied more verbatim what the current argv model is doing. I'll fix that.

Comment (bdrodes) [reply cpp/ql/src/Likely Bugs/Format/NonConstantFormat.ql]: Fixed

**Review by geoffw0 — DISMISSED** [review]
The results are much easier to understand now.  👍

**Comment by geoffw0** [comment]
> It looks like the DCA run shows no regressions? Does that mean we are set on this one?

I don't see any failures in Mathias's run, but it failed to analyze two projects.

Would be good to have a brief change note (and that'll make CI happy).

**Comment by MathiasVP** [comment]
> I don't see any failures in Mathias's run, but it failed to analyze two projects.

They both failed before the analyze step, so I wouldn't worry too much about that.

**Comment by MathiasVP** [comment]
I've added a change note now. Might as well just get this one merged ourselves given that Ben is on vacation.

**Comment by MathiasVP** [comment]
I've accepted the required changes in the PR, and I've opened an internal PR to accept the internal test changes.

**Review by MathiasVP — APPROVED** [review]

**Review by geoffw0 — APPROVED** [review]

**Pull request merged**